### PR TITLE
:sparkles: Add safety checks on object deletion

### PR DESCRIPTION
**Pull request by niwinz**

### CODE REVIEW

File: backend/src/app/migrations/sql/0117-mod-file-object-thumbnail-table.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+ALTER TABLE file_object_thumbnail
+ DROP CONSTRAINT file_object_thumbnail_file_id_fkey,
+  ADD FOREIGN KEY (file_id) REFERENCES file(id) DEFERRABLE,
+ DROP CONSTRAINT file_object_thumbnail_media_id_fkey,
+  ADD FOREIGN KEY (media_id) REFERENCES storage_object(id) DEFERRABLE;
+
+--- Mark all related storage_object row as touched
```
Comment (superalex) [inline backend/src/app/migrations/sql/0117-mod-file-object-thumbnail-table.sql:7]: Why is this code commented?

File: backend/src/app/rpc/commands/files_thumbnails.clj [hunk 2]
```diff
@@ -226,34 +227,54 @@
...
-                                ::sto/touched-at (dt/now)
+                                ::sto/touched-at tnow
                                 :content-type mtype
                                 :bucket "file-object-thumbnail"})]
 
-    (db/exec-one! conn [sql:create-object-thumbnail file-id object-id
-                        (:id media) tag (:id media)])))
+    (if (some? thumb)
+      (do
+        ;; We mark the old media id as touched if it does not matches
```
Comment (superalex) [inline backend/src/app/rpc/commands/files_thumbnails.clj:258]: "if it does not match"

File: backend/src/app/storage/gc_touched.clj [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+      SET deleted_at = now(),
+          touched_at = NULL
+    WHERE id = ANY(?::uuid[])")
+
+(defn- mark-delete-in-bulk!
+  [conn ids]
+  (let [ids (db/create-array conn "uuid" ids)]
+    (db/exec-one! conn [sql:mark-delete-in-bulk ids])))
+
+;; NOTE: A getter that retrieves the key witch will be used
```
Comment (superalex) [inline backend/src/app/storage/gc_touched.clj]: witch == "a person thought to have magic powers, especially evil ones" , wich? :D

File: backend/src/app/migrations/sql/0117-mod-file-object-thumbnail-table.sql (hunk 1 above)
Comment (niwinz) [reply backend/src/app/migrations/sql/0117-mod-file-object-thumbnail-table.sql:7]: This is the code we need to execute after release so I leave it there as commented code

**Pull request merged**